src/components/Post/Post.jsx

```diff
@@ -1,46 +1,215 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import "./Post.css";
-import { PostDropdown } from "../index";
+import { PostDropdown, Comment } from "../index";
+import { likePost, dislikePost } from "../../features/posts/postsSlice";
+import {
+  bookmarkPost,
+  removePostFromBookmark,
+} from "../../features/users/usersSlice";
+import {
+  getCommentsForPost,
+  createNewCommentToPost,
+  editCommentOfPost,
+} from "../../features/comments/commentsSlice";
+import { useDispatch, useSelector } from "react-redux";
+import TextareaAutosize from "react-textarea-autosize";
 
 function Post({ post, setIsPostEdit, setEditPostId }) {
   const { _id, content, username, name } = post;
+
+  const [isPostLikedBy, setIsPostLikedBy] = useState(false);
+  const [isPostBookmark, setIsPostBookmark] = useState(false);
   const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [showCommentBox, setShowCommentBox] = useState(false);
```

Comment:
Can be done in one state which is having object instead  of boolean directly

Reply:
Yupp actually realized later